src/data/repositories/TrackerProgramLoggerD2Repository.ts

```diff
@@ -25,6 +30,11 @@ export class TrackerProgramLoggerD2Repository implements LoggerRepository {
         this.messageTypeId = messageTypeId;
     }
 
+    // TODO: implement batchLog method
+    batchLog(_logs: Log[]): FutureData<void> {
+        return Future.success(undefined);
+    }
+
     log(log: TrackerProgramLog): FutureData<void> {
```

Comment:
It's not in this PR, but that does not look right. You are implementing a `LoggerRepository`, which has the method defined as `log(log: Log): FutureData<void>`.  But then the type here is `TrackerProgramLog`. And if it's changed to `Log`, types do not match. So something is amiss.

Reply:
Maybe can we check this in the next Technical decisions meeting?